src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkWildcardsCollector.kt

```diff
@@ -0,0 +1,50 @@
+package com.jetbrains.snakecharm.lang.psi
+
+import com.intellij.lang.injection.InjectedLanguageManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import com.jetbrains.python.psi.PyElementVisitor
+import com.jetbrains.python.psi.PyRecursiveElementVisitor
+import com.jetbrains.python.psi.PyStringLiteralExpression
+import com.jetbrains.snakecharm.stringLanguage.SmkSLFile
+import com.jetbrains.snakecharm.stringLanguage.lang.psi.elementTypes.SmkSLLanguageElement
+import com.jetbrains.snakecharm.stringLanguage.lang.psi.elementTypes.SmkSLReferenceExpressionImpl
+
+class SmkWildcardsCollector : SmkElementVisitor, PyRecursiveElementVisitor() {
+    val collectedWildcards = mutableMapOf<String, PsiElement>()
```

Comment:
Not very good API because exposes mutable map as public field. The only external usage is `collector.collectedWildcards.toList()`